src/test/java/org/sagebionetworks/bridge/services/StudyServiceTest.java

```diff
@@ -320,13 +326,16 @@ public void updateStudy() {
         assertEquals(returnedValue.getAppId(), TEST_APP_ID);
         assertEquals(returnedValue.getIdentifier(), "oneId");
         assertEquals(returnedValue.getName(), "newName");
+        assertEquals(returnedValue.getPhase(), IN_FLIGHT);
         assertNotNull(returnedValue.getCreatedOn());
         assertNotNull(returnedValue.getModifiedOn());
     }
     
-    @Test(expectedExceptions = InvalidEntityException.class)
+    @Test(expectedExceptions = EntityNotFoundException.class)
     public void updateStudyInvalidStudy() {
-        service.updateStudy(TEST_APP_ID, Study.create());
+        Study study = Study.create();
+        study.setIdentifier("oneId");
+        service.updateStudy(TEST_APP_ID, study);
```

Comment:
So, why did this change? Is this correct?

Reply:
Phase is required, so to ensure this doesn't trigger an NPE, we set it from the existing object before validation. But this requires having an identifier, which we virtually always will now that the controller sets it from the path (which is our default). So the ID is required and the exception thrown has changed.